client/components/deposit-status-pill/index.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { displayStatus } from 'deposits/strings';
+import Pill from 'components/pill';
+
+const mappings: {
+	[ key: string ]: 'primary' | 'success' | 'alert' | 'danger' | 'light';
+} = {
+	estimated: 'light',
+	pending: 'alert',
+	in_transit: 'success',
+	paid: 'success',
+	failed: 'danger',
+	canceled: 'danger',
+};
+
+/**
+ * Renders a deposits status pill.
+ * Based off of the Pill component found components/pill.
+ *
+ * @return {JSX.Element} Deposit status pill.
+ */
+const DepositStatusPill: React.FC< {
+	status: string;
+} > = ( { status } ): JSX.Element => {
+	const label = displayStatus[ status as keyof typeof displayStatus ]
+		? displayStatus[ status as keyof typeof displayStatus ]
```

Comment:
As we spoke about IRL, this `status as keyof typeof displayStatus` could be improved if the deposit status was a defined type. In the interest of moving this forward, I'm going to approve this PR and address the much wider type refactor as part of separate PR.

Reply:
New issue: https://github.com/Automattic/woocommerce-payments/issues/5987